### ISSUE TITLE
Build failure: PACKAGENAME

### ISSUE THREAD

**Issue by storopoli** [NixOS/nixpkgs#222691]
### Steps To Reproduce

1. build *sparrow*

### Build log

```
error: builder for '/nix/store/gi4ws8hh7d7r1mpl1id1v2d46j6fyh04-sparrow-modules-1.7.1.drv' failed with exit code 1;
       last 10 log lines:
       > error: auto-patchelf could not satisfy dependency libGL.so.1 wanted by /nix/store/89g7v0drv1dk749rf2jj6n1lw85c1hg3-sparrow-modules-1.7.1/modules/javafx.graphics/libprism_es2.so
       > error: auto-patchelf could not satisfy dependency libpangoft2-1.0.so.0 wanted by /nix/store/89g7v0drv1dk749rf2jj6n1lw85c1hg3-sparrow-modules-1.7.1/modules/javafx.graphics/libjavafx_font_pango.so
       > error: auto-patchelf could not satisfy dependency libpango-1.0.so.0 wanted by /nix/store/89g7v0drv1dk749rf2jj6n1lw85c1hg3-sparrow-modules-1.7.1/modules/javafx.graphics/libjavafx_font_pango.so
       > error: auto-patchelf could not satisfy dependency libgobject-2.0.so.0 wanted by /nix/store/89g7v0drv1dk749rf2jj6n1lw85c1hg3-sparrow-modules-1.7.1/modules/javafx.graphics/libjavafx_font_pango.so
       > error: auto-patchelf could not satisfy dependency libglib-2.0.so.0 wanted by /nix/store/89g7v0drv1dk749rf2jj6n1lw85c1hg3-sparrow-modules-1.7.1/modules/javafx.graphics/libjavafx_font_pango.so
       > auto-patchelf failed to find all the required dependencies.
       > Add the missing dependencies to --libs or use `--ignore-missing="foo.so.1 bar.so etc.so"`.
       > /nix/store/pw17yc3mwmsci4jygwalj8ppg0drz31v-stdenv-linux/setup: line 79: pop_var_context: head of shell_variables not a function context
       > /nix/store/pw17yc3mwmsci4jygwalj8ppg0drz31v-stdenv-linux/setup: line 1456: pop_var_context: head of shell_variables not a function context
       > /nix/store/pw17yc3mwmsci4jygwalj8ppg0drz31v-stdenv-linux/setup: line 1593: pop_var_context: head of shell_variables not a function context
       For full logs, run 'nix log /nix/store/gi4ws8hh7d7r1mpl1id1v2d46j6fyh04-sparrow-modules-1.7.1.drv'.
error: 1 dependencies of derivation '/nix/store/6qavsk22n16824bgil6k3vfq9s6i7jgf-sparrow-1.7.1.drv' failed to build
error: 1 dependencies of derivation '/nix/store/nmnv0qbckjkz438vfw4157ydshv202xc-user-environment.drv' failed to build
error: 1 dependencies of derivation '/nix/store/1cb1za6xjl36blp1nqpgcgm3wnyn5s16-etc.drv' failed to build
error: 1 dependencies of derivation '/nix/store/zk611bnixnhg548xqihlph1c1xd1vr9j-nixos-system-framework-23.05.20230321.19cf008.drv' failed to build
```

### Notify maintainers

@emmanuelrosa
@emmanuelrosa_1000101

### Metadata

Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
this path will be fetched (0.00 MiB download, 0.00 MiB unpacked):
  /nix/store/mk5p8a7rqx54qdy557chl7l5cq8kjrgl-nix-info
copying path '/nix/store/mk5p8a7rqx54qdy557chl7l5cq8kjrgl-nix-info' from 'https://cache.nixos.org'...
 - system: `"x86_64-linux"`
 - host os: `Linux 6.2.7, NixOS, 23.05 (Stoat), 23.05.20230321.19cf008`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.14.1`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`
```

**Comment by emmanuelrosa** [comment]
TLDR; This was caused by #206643 and is fixed in #221010

Upstream bundles a JDK/JRE with sparrow. Our sparrow package does not use the bundled JDK and instead replaces it with a JDK from Nixpkgs. Since the sparrow bundle contains the JDK and the sparrow code in a single JIMAGE, the sparrow package compares the files in the JIMAGE with the files provided by the JDK in Nixpkgs in order to determine which files in the JIMAGE are part of the JDK. Those files are then excluded from the sparrow derivation. 

When #206643 disabled JavaFX by default it changed the output of the aforementioned comparison, leaving behind some files which belong to the JDK as part of the sparrow derivation; The files which are part of JavaFX. This in turn causes the patchelf procedure to fail because the sparrow package is only expected to patch sparrow-specific dependencies, not the JDK.
